c++/src/kj/compat/tls.c++

```diff
@@ -312,78 +317,57 @@ private:
   ReadyInputStreamWrapper readBuffer;
   ReadyOutputStreamWrapper writeBuffer;
 
-  kj::Promise<size_t> tryReadInternal(
-      void* buffer, size_t minBytes, size_t maxBytes, size_t alreadyDone) {
-    return sslCall([this,buffer,maxBytes]() { return SSL_read(ssl, buffer, maxBytes); })
-        .then([this,buffer,minBytes,maxBytes,alreadyDone](size_t n) -> kj::Promise<size_t> {
-      if (n >= minBytes || n == 0) {
-        return alreadyDone + n;
-      } else {
-        return tryReadInternal(reinterpret_cast<byte*>(buffer) + n,
-                               minBytes - n, maxBytes - n, alreadyDone + n);
-      }
-    });
-  }
-
-  Promise<void> writeInternal(kj::ArrayPtr<const byte> first,
-                              kj::ArrayPtr<const kj::ArrayPtr<const byte>> rest) {
+  Promise<void> writeInternal(kj::ArrayPtr<const kj::ArrayPtr<const byte>> slices) {
     KJ_REQUIRE(shutdownTask == kj::none, "already called shutdownWrite()");
```

Comment:
Looks like the original code was checking for shutdownTask every recursive iteration, but now just checks at the start?

Reply:
Yeah, I think it's safer to avoid optimising while converting to coroutines in the same PR. The optimisation should come in a separate PR to avoid us missing things like this.

Reply:
> Looks like the original code was checking for shutdownTask every recursive iteration, but now just checks at the start?

My reasoning: since this code is single-threaded, shutdownWrite() can't happen while we write data out. So it should be enough calling it once?

I'm happy to move it inside the loop to if you think this is not correct.

Reply:
fwiw I tried to keep as close to the original as possible eliminating recursion.